### PR TITLE
Upgrade to Node 0.8.5. Delendency packages are also upgraded

### PR DESCRIPTION
**Pull request by prabhakhar**

### CODE REVIEW

**Comment by s3u** [comment]
I get the error when I run `make test`

test-console-sanity
Error: listen EADDRINUSE
    at errnoException (net.js:768:11)
    at HTTPServer.Server._listen2 (net.js:908:14)
    at listen (net.js:935:10)
    at HTTPServer.Server.listen (net.js:984:5)
    at Object.module.exports.sanity (/private/tmp/ql.io/modules/console/test/test-console-sanity.js:33:13)
    at Object.wrapTest (/private/tmp/ql.io/modules/console/node_modules/nodeunit/lib/core.js:235:16)
    at wrapTest (/private/tmp/ql.io/modules/console/node_modules/nodeunit/lib/core.js:235:16)
    at Object.exports.runTest (/private/tmp/ql.io/modules/console/node_modules/nodeunit/lib/core.js:69:9)
    at exports.runSuite (/private/tmp/ql.io/modules/console/node_modules/nodeunit/lib/core.js:117:25)
    at _concat (/private/tmp/ql.io/modules/console/node_modules/nodeunit/deps/async.js:508:13)
error: Error: listen EADDRINUSE
    at errnoException (net.js:768:11)
    at HTTPServer.Server._listen2 (net.js:908:14)
    at listen (net.js:935:10)
    at HTTPServer.Server.listen (net.js:984:5)
    at Object.module.exports.sanity (/private/tmp/ql.io/modules/console/test/test-console-sanity.js:33:13)
    at Object.wrapTest (/private/tmp/ql.io/modules/console/node_modules/nodeunit/lib/core.js:235:16)
    at wrapTest (/private/tmp/ql.io/modules/console/node_modules/nodeunit/lib/core.js:235:16)
    at Object.exports.runTest (/private/tmp/ql.io/modules/console/node_modules/nodeunit/lib/core.js:69:9)
    at exports.runSuite (/private/tmp/ql.io/modules/console/node_modules/nodeunit/lib/core.js:117:25)
    at _concat (/private/tmp/ql.io/modules/console/node_modules/nodeunit/deps/async.js:508:13)

**Comment by prabhakhar** [comment]
I tested previously before giving the pull request. Let me try again.

**Comment by prabhakhar** [comment]
I tried locally 3 times it passed. I setup travis build and tested. I gave 7 builds. all pass. http://travis-ci.org/#!/prabhakhar/ql.io/builds 

I will see  tomorrow  why it fails intermittently in @s3u machine.

**Comment by prabhakhar** [comment]
I am able to reproduce the issue in my old macbook. not at will, but happens time to time.
1. Reason for the failure is the previous test case is holding the port 3000 for longer time in slower machines in node 0.8. If we run test-console-sanity alone, we will never hit this issue.
2. My changes in cluster2 is not related here. Because the failed cases are in console. Only app depends on cluster2.

Solution:
1. I can change the test case to listen on a different port. Or wait until 3000 becomes free.
2. Downgrade all the dependency packages in console and see which package is causing this issue. (holding the port for longer time.) I doubt connect/express.

**Comment by prabhakhar** [comment]
I found the issue. Holding of the port is because of a bug in the test-case (that I wrote :-( ).

**Comment by prabhakhar** [comment]
Please try now. It should work. 
Make sure node 0.8.x and latest npm is installed.

Thanks to @s3u for pointing out this.

**Comment by s3u** [comment]
Yup - tested twice. No failures.
